tests/functional/lib/receptorcontrol/receptorcontrol.go

```diff
@@ -158,3 +165,152 @@ func (r *ReceptorControl) Status() (*netceptor.Status, error) {
 
 	return &status, nil
 }
+
+// WorkSubmit begins work on remote node
+func (r *ReceptorControl) WorkSubmit(node, serviceName string) (string, error) {
+	_, err := r.WriteStr(fmt.Sprintf("work submit %s %s\n", node, serviceName))
+	if err != nil {
+		return "", err
+	}
+	r.ReadStr()
+	r.CloseWrite()
+	response, err := r.ReadAndParseJSON()
+	if err != nil {
+		return "", err
+	}
+	unitID := fmt.Sprintf("%v", response["unitid"])
+	return unitID, nil
+}
+
+// WorkStart begins work on local node
+func (r *ReceptorControl) WorkStart(workID string) (string, error) {
+	_, err := r.WriteStr(fmt.Sprintf("work start %s\n", workID))
+	if err != nil {
+		return "", err
+	}
+	_, err = r.ReadStr() // flush response
+	if err != nil {
+		return "", err
+	}
+	r.CloseWrite() // clost write half to signal EOF
+	response, err := r.ReadAndParseJSON()
+	if err != nil {
+		return "", err
+	}
+	r.Close()
+	unitID := fmt.Sprintf("%v", response["unitid"])
+	return unitID, nil
+}
+
+// WorkCancel cancels work
+func (r *ReceptorControl) WorkCancel(workID string) error {
+	_, err := r.WriteStr(fmt.Sprintf("work cancel %s\n", workID))
+	if err != nil {
+		return err
+	}
+	_, err = r.ReadAndParseJSON() // flush response
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// WorkRelease cancels and deletes work
+func (r *ReceptorControl) WorkRelease(workID string) error {
+	_, err := r.WriteStr(fmt.Sprintf("work release %s\n", workID))
+	if err != nil {
+		return err
+	}
+	_, err = r.ReadAndParseJSON() // flush response
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReceptorControl) getWorkStatus(workID string) (map[string]interface{}, error) {
+	_, err := r.WriteStr(fmt.Sprintf("work status %s\n", workID))
+	if err != nil {
+		return nil, err
+	}
+	status, err := r.ReadAndParseJSON()
+	if err != nil {
+		return nil, err
+	}
+	return status, nil
+}
+
+func (r *ReceptorControl) getWorkList() (map[string]interface{}, error) {
+	_, err := r.WriteStr(fmt.Sprintf("work list\n"))
+	if err != nil {
+		return nil, err
+	}
+	workList, err := r.ReadAndParseJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	return workList, nil
+}
+
+func checkTimeout(timeout <-chan time.Time) bool {
+	select {
+	case <-timeout:
+		return true
+	default:
+		return false
+	}
+}
+
+//AssertWorkRunning waits until work status is running
+func (r *ReceptorControl) AssertWorkRunning(workID string) error {
```

Comment:
We should also make the timeouts on these `Assert` functions configurable

Reply:
okay, either we can use a context with timeout and send that in, or we can just provide a timeout option as an argument into the Assert functions.

Reply:
we might also consider an `Assert` and an `AssertWithTimeout` that just calls Assert in `CheckWithTimeout` so if you dont want to provide a context you dont have to

Reply:
> okay, either we can use a context with timeout and send that in, or we can just provide a timeout option as an argument into the Assert functions.

sorry just saw this, a context is more flexible so I think that's the right move